### ISSUE TITLE
[DOC] Exactly Once with TimeBasedPartioner

### ISSUE THREAD

**Issue by BDeus** [confluentinc/kafka-connect-storage-cloud#165]
Hi,
In file https://github.com/confluentinc/kafka-connect-storage-cloud/blob/7eec7f6ac9fb8536fd2174002fa095de3b94fb88/kafka-connect-s3/docs/s3_connector.rst#exactly-once-delivery-on-top-of-eventual-consistency 
There is this sentence "A version of time-based partitioners based only on record timestamps that will guarantee exactly-once delivery to S3 will become soon available."

Is not already been done?
It would be nice if we can find here a description of how to perform (or avoid error) exactly once semantic with TimeBasedPartitioner.

**Comment by cricket007** [comment]
See commit https://github.com/confluentinc/kafka-connect-storage-cloud/commit/60eb3b0fc21cc1431e581c9e9dab1f98ea8b85a8

**Comment by BDeus** [comment]
Yes, i already know this commit. My point was it could be nice to provide documentation about it and remove "available soon" since this commit.

**Comment by cricket007** [comment]
@niksajakovljevic I was curious - how often do you find yourself creating connects where your flush size is set to be greater than the number of messages that occur within a given partition duration? 

In other words, by using Record or RecordField extraction, the file rotation will still need to happen for the partition values, right? Anyway, I'm guessing you've set `rotate.schedule.ms` to -1 and `partition.duration.ms` to something else?

**Comment by niksajakovljevic** [comment]
@cricket007 My use case had a constant live stream of data ( 30K/s on avg). The flush size was 250K per partition and partition duration of 1h ( hourly buckets). Base on these numbers flushing is almost guaranteed to happen. Of course in the case of stopping the ingestion flow(eg. an incident) you might get increased latency in S3 but that didn't bother me as the stuff in S3 was not latency sensitive.
